### PR TITLE
Output a warning whenever user tries to use datasets without a unique key

### PR DESCRIPTION
**Pull request by BenLorantfy**
After #116, if you omit the label from your dataset because you don't need it, it fails silently and all the datasets are combined into one. 

This is a small PR to output a warning if a user tries to add datasets without keys, similar to how React outputs a warning when you omit a key.

screen shot 2017-05-23 at 8 38 26 pm

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -155,8 +155,14 @@ class ChartComponent extends React.Component {
 	  // use the key provider to work out which series have been added/removed/changed
 	  const currentDatasetKeys = currentDatasets.map(this.props.datasetKeyProvider);
 	  const nextDatasetKeys = nextDatasets.map(this.props.datasetKeyProvider);
+    const shouldWarn = !currentDatasetKeys.every(d => typeof d !== "undefined") || !nextDatasetKeys.every(d => typeof d !== "undefined");
 	  const newDatasets = nextDatasets.filter(d => currentDatasetKeys.indexOf(this.props.datasetKeyProvider(d)) === -1);
 
+    if (shouldWarn && !this.hasWarned) {
```
Comment (jerairrest) [inline src/index.js]: I believe this should only occur if you are not in a production build. :)

**Comment by kumarharsh** [comment]
+1 this should be merged!

File: src/index.js (hunk 1 above)
Comment (BenLorantfy) [inline src/index.js]: How can I check if production vs. development?

Comment (BenLorantfy) [inline src/index.js]: Seems like this will work unless you have a better method https://stackoverflow.com/a/25922668

**Comment by BenLorantfy** [comment]
Wasn't sure the best way to check the enviroment. There's this method https://stackoverflow.com/a/35470995 but that seems like it will only work if the user sets the NODE_ENV enviroment variable. There's also this method https://stackoverflow.com/a/25922668 which seems a little hacky but will probably most often. Any suggestions on how to do this would be appreciated.

**Comment by kumarharsh** [comment]
React itself uses NODE_ENV to check it. See [invariant](https://github.com/zertosh/invariant/blob/master/invariant.js)

**Comment by benmccann** [comment]
@BenLorantfy this PR will need to be rebased

**Comment by BenLorantfy** [comment]
@jerairrest , @benmccann Took me almost a year but I re-based and added the production check. How's it now? 🙂

**Comment by BenLorantfy** [comment]
@jerairrest just pinging

**Pull request merged**